### ISSUE TITLE
`Docker-compose up api` doesn't mount volume but `docker-compose run api /bin/bash` does

### ISSUE THREAD

**Issue by olalonde** [docker/compose#1517]
There's something strange going on. I used `docker-compose up api` for a few weeks and it was mounting my volumes as expected but recently it stopped mounting them. However, if I `docker-compose run api /bin/bash`, the volumes are mounted correctly. What might be causing this?

```
$ docker version
Client version: 1.5.0
Client API version: 1.17
Go version (client): go1.4.1
Git commit (client): a8a31ef
OS/Arch (client): darwin/amd64
Server version: 1.5.0
Server API version: 1.17
Go version (server): go1.4.1
Git commit (server): a8a31ef

$ boot2docker version
Boot2Docker-cli version: v1.5.0
Git commit: ccd9032
```

Here's the `docker inspect` for when I use `docker-compose up api` (edited slightly):

```json
$ docker inspect 2df29b843594
[{
...
  
        "Volumes": {
            "/src/.tmp": {},
            "/src/client": {},
            "/src/server": {},
            "/src/test": {}
        },
        "WorkingDir": "/src"
...
    "Driver": "aufs",
...
    "HostConfig": {
        "Binds": [
            "/Users/me/active/jsapp/server:/src/server:rw",
            "/Users/me/active/jsapp/.tmp:/src/.tmp:rw",
            "/Users/me/active/jsapp/test:/src/test:rw",
            "/Users/me/active/jsapp/client:/src/client:rw"
        ],
...
        "VolumesFrom": [
            "3b6f6919a1d1488264fcbb4daf6b10d17b851a6a19e85572a0cb1ec70eba6405"
        ]
    },
    "Volumes": {
        "/src/.tmp": "/mnt/sda1/var/lib/docker/vfs/dir/dd6240e392dbb5c27ce79e448438a287d3c337393d3b0b61823b6c056b5f85a2",
        "/src/client": "/mnt/sda1/var/lib/docker/vfs/dir/ae8a0f36fcf728f782fd6901fde2c726eb6e4b827ef1303e0b0dbc388ddb8b97",
        "/src/server": "/mnt/sda1/var/lib/docker/vfs/dir/07a6c5ad3e2f92f18bbdd91b0784f0a11b7e857589853c9f836434e88781a8eb",
        "/src/test": "/mnt/sda1/var/lib/docker/vfs/dir/aebea7de26889194e5e91bc2116c3d811d7c725b07088df0df384781d9ad9dae"
    },
    "VolumesRW": {
        "/src/.tmp": true,
        "/src/client": true,
        "/src/server": true,
        "/src/test": true
    }
}
]
```

Here's the `docker inspect` for when I use `docker-compose run api /bin/bash`  (edited slightly):

```json
[{
...
        "Volumes": {
            "/src/.tmp": {},
            "/src/client": {},
            "/src/server": {},
            "/src/test": {}
        },
        "WorkingDir": "/src"
...
    "Driver": "aufs",
...
    "HostConfig": {
        "Binds": [
            "/Users/me/active/jsapp/server:/src/server:rw",
            "/Users/me/active/jsapp/.tmp:/src/.tmp:rw",
            "/Users/me/active/jsapp/test:/src/test:rw",
            "/Users/me/active/jsapp/client:/src/client:rw"
        ],
        "VolumesFrom": []
    },
...
    "Volumes": {
        "/src/.tmp": "/Users/me/active/jsapp/.tmp",
        "/src/client": "/Users/me/active/jsapp/client",
        "/src/server": "/Users/me/active/jsapp/server",
        "/src/test": "/Users/me/active/jsapp/test"
    },
    "VolumesRW": {
        "/src/.tmp": true,
        "/src/client": true,
        "/src/server": true,
        "/src/test": true
    }
}
]
```

**Comment by dnephin** [comment]
I think this has to do with preserving volumes on recreate. There is a fix for this in the 1.3.0 rc. 

I'd try running `docker-compose rm` to clear out the old containers first.

**Comment by olalonde** [comment]
This fixed it.
